Initialize composable observable store after update

Gudahtt opened the pull request:
## Explanation

The composable observable store now updates state immediately when the structure is updated. Previously each store would only be updated after the first state change. This ensures that the composable observable store state is always complete.

## Manual Testing Steps

This PR should ensure that the background state and the UI state contains an entry for every single controller immediately upon construction, without needing to wait for each controller to update. This is most obvious by looking at the background state snapshot captured by the `errors.spec.js` test suite. You can manually test this by looking at the background state in the background console shortly after the extension starts (e.g. using breakpoints or one of the state hooks).

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

Gudahtt committed:
Initialize composable observable store after update

The composable observable store now updates state immediately when the
structure is updated. Previously each store would only be updated after
the first state change. This ensures that the composable observable
store state is always complete.

Gudahtt commented:
This depends upon #20465 and #20457

FrederikBolding commented on app/scripts/lib/ComposableObservableStore.js:
```diff
@@ -72,7 +73,10 @@ export default class ComposableObservableStore extends ObservableStore {
         );
       }
+
+      initialState[key] = store.state || store.getState?.();
```
```suggestion
      initialState[key] = store.state ?? store.getState?.();
```
Should this be a nullish coalescing operator instead?

Gudahtt replied:
Good point, yes I think it should. I don't think we have any examples of stores with a top-level falsy state, but I see no reason to assume that won't happen.

Gudahtt replied:
Test added here: f3b82b45534b1c4e0ed6a432f2d42179cf39aed7

Gudahtt committed:
SUpport falsy controller state

We now use the nullish coalescing operator when checkint store.state, so that we don't accidentally ignore falsy state.

Co-authored-by: Frederik Bolding <frederik.bolding@gmail.com>

Gudahtt committed:
Add test for falsy controller state

Gudahtt committed:
Update state snapshots

A change on `develop` required another state update.

mcmire reviewed (approved):
Makes sense!
